modules/cluster/main.tf

```diff
@@ -128,7 +128,8 @@ module "eks" {
   node_groups = !var.enable_worker_group ? local.node_groups_extended : {}
 
   workers_additional_policies = [
-    "arn:${data.aws_partition.current.partition}:iam::aws:policy/AmazonEC2ContainerRegistryPowerUser"
+    "arn:${data.aws_partition.current.partition}:iam::aws:policy/AmazonEC2ContainerRegistryPowerUser",
+    var.enable_ebscsi_driver ? "arn:${data.aws_partition.current.partition}:iam::aws:policy/service-role/AmazonEBSCSIDriverPolicy" : ""
```

Comment:
Don't add the permissions to the workers. Instead an IAM role should be created (see irsa.tf) and attached to the addon using service_account_role_arn.

Reply:
I can work on this.